src/Intervals.php

```diff
@@ -127,12 +127,12 @@ public static function get(ConstraintInterface $constraint)
     }
 
     /**
-     * @phpstan-return array{'intervals': array<int, array{'start': Constraint, 'end': Constraint}>, 'devConstraints': Constraint[]}
+     * @phpstan-return array{'intervals': Interval[], 'devConstraints': Constraint[]}
      */
     private static function generateIntervals(ConstraintInterface $constraint, $stopOnFirstValidInterval = false)
     {
         if ($constraint instanceof EmptyConstraint) {
-            return array('intervals' => array(array('start' => self::zero(), 'end' => self::positiveInfinity())), 'devConstraints' => array(self::anyDev()));
+            return array('intervals' => array(new Interval(Interval::zero(), Interval::positiveInfinity())), 'devConstraints' => array(Interval::anyDev()));
```

Comment:
Imho we have the same optimization possibility here as we'd have with a `ConstraintFactory`: we don't need a new instance of `Interval` over and over again if the interval is exactly the same. We could reuse existing interval instances, right? :)

Reply:
Sure, but IMO that's still best done with the memoization on get(), because the expensive stuff here is computing intervals for a constraint, not storing them. Especially as they'll get used at poolbuilder time and then it can be cleared as it becomes useless. Feel free to PR if you have time though :P